lib/codegen_types.re

```diff
@@ -65,39 +73,55 @@ let codegen_rule_body = (rule_body: B.rule_body): string => {
    }
 }
 
-let codegen_rules = (rules: list(B.rule)): (list(string)) => {
-   let rules_sorted_by_type = List.sort(
-      ((a_i, a_body): B.rule, (b_i, b_body): B.rule) => {
-         if (a_i == "program") {
-            /* make sure program is at last */
-            1;
-         } else if (b_i == "program") {
-            -1;
-         } else {
-            let a_type = B.show_rule_body(a_body);
-            let b_type = B.show_rule_body(b_body);
-            /* make sure leaves are at top */
-            switch(a_body) {
-            | B.SIMPLE(B.ATOM(B.TOKEN)) => -1
-            | _ => {
-               switch(b_body) {
-               | B.SIMPLE(B.ATOM(B.TOKEN)) => 1
-               | _ => String.compare(a_type, b_type)
-               }
-            }
-            }
+let sort_by_type = ((a_i, a_body): B.rule, (b_i, b_body): B.rule) => {
+   if (a_i == "program") {
```

Comment:
You should use instead the name of the entry point (it's the first element in ast_grammar_normalized.grammar)